new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/Domain.java

```diff
@@ -292,4 +324,231 @@ public static boolean scaleClusterWithRestApi(String domainUid,
+   *
+   * @param clusterName name of the WebLogic cluster to be scaled in the domain
+   * @param domainUid domainUid of the domain to be scaled
+   * @param domainNamespace domain namespace in which the domain exists
+   * @param domainHomeLocation domain home location of the domain
+   * @param scalingAction scaling action, accepted value: scaleUp or scaleDown
+   * @param scalingSize number of servers to be scaled up or down
+   * @param opNamespace namespace of WebLogic operator
+   * @param opServiceAccount service account of operator
+   * @param myWebAppName web app name deployed to the domain used in the WLDF policy expression
+   * @param curlCommand curl command to call the web app used in the WLDF policy expression
+   * @return true if scaling the cluster succeeds, false otherwise
+   * @throws ApiException if Kubernetes client API call fails
+   * @throws IOException if an I/O error occurs
+   * @throws InterruptedException if any thread has interrupted the current thread
+   */
+  public static boolean scaleClusterWithWLDF(String clusterName,
+                                             String domainUid,
+                                             String domainNamespace,
+                                             String domainHomeLocation,
+                                             String scalingAction,
+                                             int scalingSize,
+                                             String opNamespace,
+                                             String opServiceAccount,
+                                             String myWebAppName,
+                                             String curlCommand)
+      throws ApiException, IOException, InterruptedException {
+
+    // create RBAC API objects for WLDF script
+    logger.info("Creating RBAC API objects for WLDF script");
+    if (!createRbacApiObjectsForWLDFScript(domainNamespace, opNamespace)) {
+      return false;
+    }
+
+    // copy scalingAction.sh to Admin Server pod
+    // NOTE: you must copy scalingAction.sh to $DOMAIN_HOME/bin/scripts on admin server pod
+    String adminServerPodName = domainUid + "-" + ADMIN_SERVER_NAME_BASE;
+    V1Pod adminPod = Kubernetes.getPod(domainNamespace, null, adminServerPodName);
+    if (adminPod == null) {
+      logger.info("The admin pod {0} does not exist in namespace {1}!", adminServerPodName, domainNamespace);
+      return false;
+    }
+
+    // create $DOMAIN_HOME/bin/scripts directory on admin server pod
+    logger.info("Creating directory {0}/bin/scripts on admin server pod", domainHomeLocation);
+    ExecResult result = exec(adminPod, null, true,
+        "/bin/sh", "-c", "mkdir -p " + domainHomeLocation + "/bin/scripts");
+    if (result.exitValue() != 0) {
+      return false;
+    }
+
+    logger.info("Copying scalingAction.sh to admin server pod");
+    try {
+      copyFileToPod(domainNamespace, adminServerPodName, null,
```

Comment:
This feels like it is working around a bug. Why do we have to copy a script to the admin pod?

Reply:
From the doc: https://oracle.github.io/weblogic-kubernetes-operator/userguide/managing-domains/domain-lifecycle/scaling/#using-a-wldf-policy-rule-and-script-action-to-call-the-operators-rest-scale-api
This is required to copy scalingAction.sh to admin server pod.